connect/src/com/telenor/connect/id/ConnectIdService.java

```diff
@@ -170,6 +175,42 @@ public void failure(RetrofitError error) {
         }
     }
 
+    public void logOut(Context context) {
+        final String accessToken = getAccessToken();
+        if (accessToken == null) {
+            Log.w(ConnectUtils.LOG_TAG, "Trying to log out when user is already logged out.");
+            return;
+        }
+
+        Date accessTokenExpirationTime = getAccessTokenExpirationTime();
+        Date now = new Date();
+
+        boolean accessTokenHasExpired = now.after(accessTokenExpirationTime);
+        if (accessTokenHasExpired) {
```

Comment:
This check does not take into consider that the access token may have been revoked in other ways, or that the refresh token may still be valid. I think a safer approach would be always to try to refresh the access token first. If the revoke call is successful, it is very likely that the new access token will be valid immediately after. If revoke fails, the client does not have access anymore, so it can be considered logged out.